src/components/GraphCard/GraphCard.tsx

```diff
@@ -26,32 +27,41 @@ export default function GraphCard({
   graphName,
 }: GraphCardProps) {
   const { t } = useTranslation()
+  const [loading] = useState(false)
+
   return (
-    graphData && (
-      <Card
-        {...(onClick ? { onClick: onClick } : {})}
-        classes={{
-          root: styles[`graph-card--${open ? 'open' : 'closed'}`],
-        }}
-      >
-        <div className={styles[`labels-container--${open ? 'open' : 'closed'}`]}>
-          <Typography className={styles['region-label']}>{t(region)}</Typography>
-          <Typography className={styles['graph-label']}>{t(graphName)}</Typography>
-        </div>
-        {open && (
-          <Plot
-            data={graphData}
-            className={styles['graph-plots']}
-            config={plotlyTheme.config}
-            layout={{
-              ...plotlyTheme.layout,
-              yaxis: { title: { text: t(yAxisTitle), ...plotlyTheme.layout.yaxis.title } },
-              xaxis: { title: { text: t(xAxisTitle), ...plotlyTheme.layout.xaxis.title } },
-            }}
-            style={{ width: '100%', height: '100%' }}
-          />
-        )}
-      </Card>
-    )
+    <Card
+      {...(onClick ? { onClick: onClick } : {})}
+      classes={{
+        root: styles[`graph-card--${open ? 'open' : 'closed'}`],
+      }}
+    >
+      <div className={styles[`labels-container--${open ? 'open' : 'closed'}`]}>
+        <Typography className={styles['region-label']}>{t(`regions.${region}`)}</Typography>
+        <Typography className={styles['graph-label']}>{t(graphName)}</Typography>
+      </div>
+      {loading ? (
+        open && <LoadingState height={200} />
+      ) : open && graphData !== null ? (
+        <Plot
+          data={graphData}
+          className={styles['graph-plots']}
+          config={plotlyTheme.config}
+          layout={{
+            ...plotlyTheme.layout,
+            yaxis: { title: { text: t(yAxisTitle), ...plotlyTheme.layout.yaxis.title } },
+            xaxis: { title: { text: t(xAxisTitle), ...plotlyTheme.layout.xaxis.title } },
+          }}
+          style={{ width: '100%', height: '100%' }}
+        />
+      ) : (
+        open && (
+          // Temporary component --> To be completed in C103
+          <div style={{ padding: '100px 10px 0', height: '200px' }}>
+            {t('graphs.no_data_available')}
+          </div>
+        )
+      )}
+    </Card>
```

Comment:
just a heads up: the condition here is looking messy, I would refactor this in  #21